src/lib.rs

```diff
@@ -1078,10 +1072,26 @@ impl<T: PartialEq> ThinVec<T> {
 }
 
 impl<T> Drop for ThinVec<T> {
+    #[inline]
     fn drop(&mut self) {
-        unsafe {
-            ptr::drop_in_place(&mut self[..]);
-            self.deallocate();
+        #[cold]
+        #[inline(never)]
+        fn drop_non_singleton<T>(this: &mut ThinVec<T>) {
+            unsafe {
+                ptr::drop_in_place(&mut this[..]);
+
+                #[cfg(feature = "gecko-ffi")]
+                if !this.ptr.as_ref().uses_stack_allocated_buffer() {
```

Comment:
Woah, I didn't realize you could cfg if blocks to actually allow the code to not exist! Nice.

Reply:
That said, couldn't this `!` be removed and and this turned into an early-return? Then we don't duplicated the dealloc.

Reply:
Good point.